installer/app/src/views/modal/cluster-delete.js.jsx

```diff
@@ -10,8 +10,11 @@ var ClusterDelete = React.createClass({
 					<header>
 						<h2>Delete cluster {this.props.clusterID}?</h2>
 					</header>
-
-					<button type="submit" style={RedBtnCSS}>Delete Cluster</button>
+					<label>
+						<div>Type the cluster ID ({this.props.clusterID}) to confirm deletion</div>
+						<input type="text" value={this.state.confirmation} onChange={this.handleConfirmation} />
+					</label>
+					<button type="submit" style={RedBtnCSS} disabled={this.state.confirmation !== this.props.clusterID}>Delete Cluster</button>
```

Comment:
This needs styling when disabled. (You can import `disabled as DisabledBtnCSS` with the other button css imports at above, put the disabled condition into a boolean, `import { extend } from 'marbles/utils'`, then do something like `style={extend({}, RedBtnCSS, submitDisabled ? DisabledBtnCSS : {})} disabled={submitDisabled}`).